lib/core/base/virtual-node/serial-virtual-node.js

```diff
@@ -52,11 +52,25 @@ function normaliseProps(serialNode) {
 		`nodeName has to be a string, got '${nodeName}'`
 	);
 
+	nodeName = nodeName.toLowerCase();
+	let type = null;
+	if (nodeName === 'input') {
+		type = (
+			serialNode.type ||
+			(serialNode.attributes && serialNode.attributes.type) ||
```

Comment:
I think we should check that type actually exists. `type="foo"` should be `props.type = 'text'`.